libres/lib/analysis/update.cpp

```diff
@@ -477,12 +501,12 @@ void copy_parameters(enkf_fs_type *source_fs, enkf_fs_type *target_fs,
     }
 }
 
-void assert_size_equal(int ens_size, const bool_vector_type *ens_mask) {
-    if (bool_vector_size(ens_mask) != ens_size)
+void assert_size_equal(int ens_size, const std::vector<bool> &ens_mask) {
+    if (ens_mask.size() != ens_size)
         throw std::logic_error(
             "fundamental inconsistency detected. Total ens_size:" +
             std::to_string(ens_size) +
-            ", mask_size:" + std::to_string(bool_vector_size(ens_mask)));
+            ", mask_size:" + std::to_string(ens_mask.size()));
```

Comment:
Here I would suggest using the fmt library:
```
auto error_msg = fmt::format("fundamental.... ens_size:{}, mask_size: {}", ens_size, ens_mask.size());
```

Reply:
Dont disagree, but not related to this PR, which is already touching a lot of lines, so implemented it here: https://github.com/equinor/ert/pull/2974